### ISSUE TITLE
df ignored question, are sonar.oe.database and sonar.sources related

### ISSUE THREAD

**Issue by ccecvb** [Riverside-Software/sonar-openedge#1078]
We have been reorganizing some of our repos, and I'm now revising our scanner config.

# TL/DR

Is there a relation between the df in `sonar.oe.database` and the df found in the `sonar.sources`  ?

Or can I use `sonar.oe.database`  to specify databases required by the source analyses and `sonar.sources` for df files I want analysed separately ?

# Full story

Our main database is shared among different projects.
Some projects have additional databases.

I generated a df directory containing all df files during the build, but I now notice these df files get ignored by the sonar-scanner.

`WARN: File 'X:\cce\build\develop\localdata\df\LISABI.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.`

but that database is listed in

`sonar.oe.databases=../../localdata/df/LISABI.df,../../localdata/df/LISA.df`

This gives me the impression that the df files in the sources and the ones in databases are unrelated. Is this correct ?

If that is correct, I'd like to generate separate df files in each project containing only the tables that belong to that repo

### Trimmed scan log

```
X:\cce\build\develop\git\lisabi>C:\apps\sonar-scanner-5.0.1.3006-windows\bin\sonar-scanner.bat -Dproject.settings=sonar.properties
INFO: Scanner configuration file: C:\apps\sonar-scanner-5.0.1.3006-windows\bin\..\conf\sonar-scanner.properties
INFO: Project root configuration file: X:\cce\build\develop\git\lisabi\sonar.properties
INFO: SonarScanner 5.0.1.3006
INFO: Java 17.0.7 Eclipse Adoptium (64-bit)
INFO: Windows Server 2022 10.0 amd64
INFO: Analyzing on SonarQube server 9.9.1.69595
INFO: Default locale: "en_US", source code encoding: "UTF-8"

INFO: Project key: CCE_Develop:lisabi
INFO: Base dir: X:\cce\build\develop\git\lisabi
INFO: Working dir: X:\cce\build\develop\git\lisabi\.scannerwork
INFO: Load project settings for component key: 'CCE_Develop:lisabi'

INFO: Project configuration:
INFO:   Excluded sources: **/cceinc/**, **/utils/**, **/conv/**
INFO:   Excluded sources for duplication: **/*.w  # Exclude all .w, too many FP
WARN: File 'X:\cce\build\develop\localdata\df\AUTH.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\BIBLIOTHEEK.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\DEVINFO.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\LISA.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\LISABI.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\LISADMS.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\LISALRAP.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\MQDB.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
WARN: File 'X:\cce\build\develop\localdata\df\SmartDB.df' is ignored. It is not located in project basedir 'X:\cce\build\develop\git\lisabi'.
INFO: 5 files indexed
INFO: 0 files ignored because of inclusion/exclusion patterns
INFO: 0 files ignored because of scm ignore settings
INFO: Quality profile for oe: CceSmartTools
INFO: ------------- Run sensors on module CCE_Develop Lisa BI
INFO: Load metrics repository
INFO: Load metrics repository (done) | time=43ms
INFO: Sensor JaCoCo XML Report Importer [jacoco]
INFO: 'sonar.coverage.jacoco.xmlReportPaths' is not defined. Using default locations: target/site/jacoco/jacoco.xml,target/site/jacoco-it/jacoco.xml,build/reports/jacoco/test/jacocoTestReport.xml
INFO: No report imported, no coverage information will be imported by JaCoCo XML Report Importer
INFO: Sensor JaCoCo XML Report Importer [jacoco] (done) | time=6ms
INFO: Sensor CSS Rules [javascript]
INFO: No CSS, PHP, HTML or VueJS files are found in the project. CSS analysis is skipped.
INFO: Sensor CSS Rules [javascript] (done) | time=2ms
INFO: Sensor C# Project Type Information [csharp]
INFO: Sensor C# Project Type Information [csharp] (done) | time=2ms
INFO: Sensor C# Analysis Log [csharp]
INFO: Sensor C# Analysis Log [csharp] (done) | time=34ms
INFO: Sensor C# Properties [csharp]
INFO: Sensor C# Properties [csharp] (done) | time=0ms
INFO: Sensor OpenEdgeCodeColorizer [cabllicense]
INFO: OpenEdge plugin version: 2.22.3-941b9134
INFO: DLC directory 'c:\progress\oe122' will be added to PROPATH
INFO: Using PROPATH : ../ccesmarttools/CceFramework,../ccesmarttools/CceFramework/src,../lisaerp/appl,../smartcomponentlibrary,../smartcomponentlibrary/src,../lisafin/applfin/src,../lisaerp/appl/src,src,../ccesmarttools/CceFramework/src/devtools
INFO: Using schema : ../../localdata/df/LISABI.df,../../localdata/df/LISA.df
INFO: 18888 RCode read in 8816 ms - 11955 classes - 73541 methods - 62927 properties
INFO: Looking for submodule #1
INFO:   No pattern found - Leaving...
INFO: Sensor OpenEdgeCodeColorizer [cabllicense] (done) | time=11916ms
INFO: Sensor OpenEdgeSensor [cabllicense]
INFO: Sensor OpenEdgeSensor [cabllicense] (done) | time=31ms
INFO: Sensor OpenEdgeWarningsSensor [cabllicense]
INFO: 5 warning files imported
INFO: Sensor OpenEdgeWarningsSensor [cabllicense] (done) | time=2ms
INFO: Sensor OpenEdgeProparseSensor [cabllicense]
INFO: File src/lisabi/opbouwenlisabi-s.w - Switching to LL prediction mode - Token: ["DEF-PR-NR",<ID>,macro=6005,start=55:544:107,end=55:544:115]
INFO: 5 files proparse'd, 3 XREF files, 3 listing files, 0 failure(s), 17073 NCLOCs
INFO: AST Generation | time=16844 ms
INFO: XREF Parsing   | time=458 ms
INFO: Rules          | time=8682 ms
INFO: Sensor OpenEdgeProparseSensor [cabllicense] (done) | time=26754ms
INFO: Sensor HTML [web]
INFO: Sensor HTML [web] (done) | time=4ms
INFO: Sensor TextAndSecretsSensor [text]
INFO: 5 source files to be analyzed
INFO: 5/5 source files have been analyzed
INFO: Sensor TextAndSecretsSensor [text] (done) | time=439ms
INFO: Sensor VB.NET Project Type Information [vbnet]
INFO: Sensor VB.NET Project Type Information [vbnet] (done) | time=1ms
INFO: Sensor VB.NET Analysis Log [vbnet]
INFO: Sensor VB.NET Analysis Log [vbnet] (done) | time=21ms
INFO: Sensor VB.NET Properties [vbnet]
INFO: Sensor VB.NET Properties [vbnet] (done) | time=0ms
INFO: Sensor AnalyticsSensor [cabllicense]
INFO: Sensor AnalyticsSensor [cabllicense] (done) | time=113ms
INFO: Sensor LicenseWarningSensor [cabllicense]
INFO: Sensor LicenseWarningSensor [cabllicense] (done) | time=1ms
INFO: Sensor IaC Docker Sensor [iac]
INFO: 0 source files to be analyzed
INFO: 0/0 source files have been analyzed
INFO: Sensor IaC Docker Sensor [iac] (done) | time=101ms
INFO: ------------- Run sensors on project
INFO: Sensor Analysis Warnings import [csharp]
INFO: Sensor Analysis Warnings import [csharp] (done) | time=1ms
INFO: Sensor Zero Coverage Sensor
INFO: Sensor Zero Coverage Sensor (done) | time=2ms
INFO: CPD Executor Calculating CPD for 5 files
INFO: CPD Executor CPD calculation finished (done) | time=118ms
INFO: Analysis report generated in 180ms, dir size=1.8 MB
INFO: Analysis report compressed in 136ms, zip size=308.4 kB
INFO: Analysis report uploaded in 66ms
INFO: ------------------------------------------------------------------------
INFO: EXECUTION SUCCESS
INFO: ------------------------------------------------------------------------
INFO: Total time: 49.082s
INFO: Final Memory: 114M/704M
INFO: ------------------------------------------------------------------------

```

**Comment by ccecvb** [comment]
Tested on a small project, my test seems to confirm they are unrelated. df in sonar.source is analyzed and the one in sonar.oe.database

**Comment by gquerret** [comment]
Right, they are unrelated. The content of `sonar.oe.databases` is only by the parser so that it knows about the table / fields / indexes. If you want to execute rules on DF, then they have to be part of `sonar.sources`. You can also use `sonar.inclusions` and `sonar.exclusions` to filter file names.

**Comment by ccecvb** [comment]
Thank you for the info
